### ISSUE TITLE
Minetweaker support for Compaction is not registered, doesn't work

### ISSUE THREAD

**Issue by keithaw** [jaquadro/StorageDrawers#423]
The Compaction function is not recognized in crafttweaker scripts.

**Comment by jaquadro** [comment]
What version?  What script did you try to run?  Do the blacklist/whitelist functions work?

**Comment by keithaw** [comment]
1.10.2-3.5.17
I did not attempt the whitelist or blacklist functions. 
MT log says "No such member:Compaction"

I'm having no luck with the compactingRules config, either.

**Comment by jaquadro** [comment]
Show me the actual lines you tried (same for the config file)

**Comment by keithaw** [comment]
storagedrawers.zs:
mods.storagedrawers.Compaction.add(<minecraft:nether_wart_block>, <minecraft:nether_wart>, 9);
mods.storagedrawers.Compaction.add(<earthworks:block_slate>, <minecraft:flint>, 4);

StorageDrawers.cfg:
...
registries {
    # Items should be in form domain:item or domain:item:meta. [default: [minecraft:clay, minecraft:clay_ball, 4]]
    S:compactingRules <
        earthworks:block_slate,minecraft:flint,4
        minecraft:nether_wart_block,minecraft:nether_wart,9
     >

    # List of ore dictionary names to blacklist for substitution. [default: ]
    S:oreBlacklist <
     >

    # List of ore dictionary names to whitelist for substitution. [default: ]
    S:oreWhitelist <
     >
    B:registerExtraCompactingRules=true
}

**Comment by keithaw** [comment]
I notice in integration/minetweaker.java, something appears to be missing:

15     @Override 
16     public void init () throws Throwable { 
17         MineTweakerAPI.registerClass(OreDictionaryBlacklist.class); 
18         MineTweakerAPI.registerClass(OreDictionaryWhitelist.class); 
19     }

**Comment by jaquadro** [comment]
This  might be a dumb parsing oversight, but does the config work if you put spaces after the commas?  Here's a ruleset in use on my server:

```
    S:compactingRules <
        minecraft:clay, minecraft:clay_ball, 4
        minecraft:nether_wart_block, minecraft:nether_wart, 9
        minecraft:magma, minecraft:magma_cream, 4
        minecraft:red_sandstone, minecraft:sand:1, 4
        minecraft:prismarine, minecraft:prismarine_shard, 4
     >
```

For MT, I guess I forgot to register that one.

**Comment by keithaw** [comment]
That works, and I was able to add the flint to slate recipe as well.  One surprise is that I have to completely exit out of minecraft (not just the current world) in order for the new settings to get picked up.  The Mod Options menu does not make that obvious in the same way it does for other options.

**Comment by jaquadro** [comment]
The MT function is fixed in 3.6.0.  I've also fixed the config so it updates the registries if you change them mid-game.

**Comment by keithaw** [comment]
Awesome! Thanks.
